templates/telemetry_derived/fxa_users_services_last_seen_v1/query.sql

```diff
@@ -0,0 +1,78 @@
+WITH
+  _current AS (
+  SELECT
+    -- In this raw table, we capture the history of activity over the past
+    -- 28 days for each usage criterion as a single 64-bit integer. The
+    -- rightmost bit represents whether the user was active in the current day.
+    CAST(TRUE AS INT64) AS days_seen_bits,
+    -- Record days on which the user was in a "Tier 1" country;
+    -- this allows a variant of country-segmented MAU where we can still count
+    -- a user that appeared in one of the target countries in the previous
+    -- 28 days even if the most recent "country" value is not in this set.
+    CAST(seen_in_tier1_country AS INT64) AS days_seen_in_tier1_country_bits,
+    CAST(registered AS INT64) AS days_registered_bits,
+    * EXCEPT (submission_date, seen_in_tier1_country, registered)
+  FROM
+    `moz-fx-data-shared-prod.telemetry_derived.fxa_users_services_daily_v1`
+  WHERE
+    submission_date = @submission_date ),
+  --
+  _previous AS (
+  SELECT
+    * EXCEPT (submission_date, resurrected_same_service, resurrected_any_service)
+  FROM
+    `moz-fx-data-shared-prod.telemetry_derived.fxa_users_services_last_seen_v1`
+  WHERE
+    submission_date = DATE_SUB(@submission_date, INTERVAL 1 DAY)
+    -- Filter out rows from yesterday that have now fallen outside the 28-day window.
+    AND udf_shift_bits_one_day(days_seen_bits) > 0),
+  --
+  combined AS (
+  SELECT
+    @submission_date AS submission_date,
+  IF
+    (_current.user_id IS NOT NULL,
+      _current,
+      _previous).* REPLACE ( --
+      udf_combine_adjacent_days_bits(_previous.days_seen_bits,
+        _current.days_seen_bits) AS days_seen_bits,
+      udf_combine_adjacent_days_bits(_previous.days_seen_in_tier1_country_bits,
+        _current.days_seen_in_tier1_country_bits) AS days_seen_in_tier1_country_bits,
+      udf_coalesce_adjacent_days_bits(_previous.days_registered_bits,
+        _current.days_registered_bits) AS days_registered_bits ),
+    _previous.user_id IS NULL AS new_observation -- used for determining "resurrected" criteria
```

Comment:
Please take an early look at the "resurrected" logic below. Basically, if we don't have a "previous" entry, it means the user hasn't been seen for at least 28 days and has fallen off the radar. We use joins below to check if each user has been seen before.

Reply:
This seems to make sense to me. One ambiguity (that we may have to live with) that will result from this is that existing users who have not had any activity since before 2019-03-01 (and therefore will not be present in      fxa_users_services_first_seen_v1) will not be counted as resurrected. I don't know how many users this will affect, but it will lead to us under-counting the true number of resurrected users.

Again, I think this is the best we can do, but it may be worth documenting.